nplab/instrument/spectrometer/__init__.py

```diff
@@ -691,9 +691,8 @@ def continuously_update_spectrum(self):
 
     def update_display(self, spectrum):
         #Update the graphs
-        spectrum = spectrum[~np.isnan(spectrum)]
-        wavelengths = self.spectrometer.wavelengths[~np.insnan(spectrum)]
-        
+        spectrum = np.array([[0 if np.isnan(i) else i for i in s] for s in self.spectrometer.spectrum])
```

Comment:
Just wondering, if you're going to do that kind of nested loop to retain the multispectrometer functionality of `spectrum`, I wonder if it would make sense to do `spectrum = [s[~np.isnan(s)] for s in spectrum]; wavelengths = [w[~np.isnan(s)] for s, w in zip(spectrum, wavelengths)]`

Reply:
nvm, you just did it